In [ ]:
if (!require("dplyr")) install.packages("dplyr")
install.packages('nycflights13')
#Librería de la BD
library(nycflights13)
library(dplyr)

Loading required package: dplyr


Attaching package: ‘dplyr’


The following objects are masked from ‘package:stats’:

    filter, lag


The following objects are masked from ‘package:base’:

    intersect, setdiff, setequal, union


Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



In [ ]:
head(flights, 30) #air_time	distance
#head(weather) #Intresting humid, wind_dir,	wind_speed, temp
#head(planes) #X
#head(airports) #Xhead(airlines) #X

year,month,day,dep_time,sched_dep_time,dep_delay,arr_time,sched_arr_time,arr_delay,carrier,flight,tailnum,origin,dest,air_time,distance,hour,minute,time_hour
<int>,<int>,<int>,<int>,<int>,<dbl>,<int>,<int>,<dbl>,<chr>,<int>,<chr>,<chr>,<chr>,<dbl>,<dbl>,<dbl>,<dbl>,<dttm>
2013,1,1,517,515,2,830,819,11,UA,1545,N14228,EWR,IAH,227,1400,5,15,2013-01-01 05:00:00
2013,1,1,533,529,4,850,830,20,UA,1714,N24211,LGA,IAH,227,1416,5,29,2013-01-01 05:00:00
2013,1,1,542,540,2,923,850,33,AA,1141,N619AA,JFK,MIA,160,1089,5,40,2013-01-01 05:00:00
2013,1,1,544,545,-1,1004,1022,-18,B6,725,N804JB,JFK,BQN,183,1576,5,45,2013-01-01 05:00:00
2013,1,1,554,600,-6,812,837,-25,DL,461,N668DN,LGA,ATL,116,762,6,0,2013-01-01 06:00:00
2013,1,1,554,558,-4,740,728,12,UA,1696,N39463,EWR,ORD,150,719,5,58,2013-01-01 05:00:00
2013,1,1,555,600,-5,913,854,19,B6,507,N516JB,EWR,FLL,158,1065,6,0,2013-01-01 06:00:00
2013,1,1,557,600,-3,709,723,-14,EV,5708,N829AS,LGA,IAD,53,229,6,0,2013-01-01 06:00:00
2013,1,1,557,600,-3,838,846,-8,B6,79,N593JB,JFK,MCO,140,944,6,0,2013-01-01 06:00:00


In [ ]:
str(flights)

tibble [336,776 × 19] (S3: tbl_df/tbl/data.frame)
 $ year          : int [1:336776] 2013 2013 2013 2013 2013 2013 2013 2013 2013 2013 ...
 $ month         : int [1:336776] 1 1 1 1 1 1 1 1 1 1 ...
 $ day           : int [1:336776] 1 1 1 1 1 1 1 1 1 1 ...
 $ dep_time      : int [1:336776] 517 533 542 544 554 554 555 557 557 558 ...
 $ sched_dep_time: int [1:336776] 515 529 540 545 600 558 600 600 600 600 ...
 $ dep_delay     : num [1:336776] 2 4 2 -1 -6 -4 -5 -3 -3 -2 ...
 $ arr_time      : int [1:336776] 830 850 923 1004 812 740 913 709 838 753 ...
 $ sched_arr_time: int [1:336776] 819 830 850 1022 837 728 854 723 846 745 ...
 $ arr_delay     : num [1:336776] 11 20 33 -18 -25 12 19 -14 -8 8 ...
 $ carrier       : chr [1:336776] "UA" "UA" "AA" "B6" ...
 $ flight        : int [1:336776] 1545 1714 1141 725 461 1696 507 5708 79 301 ...
 $ tailnum       : chr [1:336776] "N14228" "N24211" "N619AA" "N804JB" ...
 $ origin        : chr [1:336776] "EWR" "LGA" "JFK" "JFK" ...
 $ dest          : ch

In [ ]:
summary(flights)

      year          month             day           dep_time    sched_dep_time
 Min.   :2013   Min.   : 1.000   Min.   : 1.00   Min.   :   1   Min.   : 106  
 1st Qu.:2013   1st Qu.: 4.000   1st Qu.: 8.00   1st Qu.: 907   1st Qu.: 906  
 Median :2013   Median : 7.000   Median :16.00   Median :1401   Median :1359  
 Mean   :2013   Mean   : 6.549   Mean   :15.71   Mean   :1349   Mean   :1344  
 3rd Qu.:2013   3rd Qu.:10.000   3rd Qu.:23.00   3rd Qu.:1744   3rd Qu.:1729  
 Max.   :2013   Max.   :12.000   Max.   :31.00   Max.   :2400   Max.   :2359  
                                                 NA's   :8255                 
   dep_delay          arr_time    sched_arr_time   arr_delay       
 Min.   : -43.00   Min.   :   1   Min.   :   1   Min.   : -86.000  
 1st Qu.:  -5.00   1st Qu.:1104   1st Qu.:1124   1st Qu.: -17.000  
 Median :  -2.00   Median :1535   Median :1556   Median :  -5.000  
 Mean   :  12.64   Mean   :1502   Mean   :1536   Mean   :   6.895  
 3rd Qu.:  11.00   3rd Qu.:1

**COCIENTE DE VARIANZAS**

In [ ]:
#Instalación del paquete stests
if (!require('devtools')) install.packages('devtools')
devtools::install_github('fhernanb/stests', force=TRUE)

Loading required package: devtools

Loading required package: usethis




HDInterval (NA -> 0.2.4) [CRAN]
expm       (NA -> 1.0-0) [CRAN]


Installing 2 packages: HDInterval, expm

Installing packages into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



── R CMD build ─────────────────────────────────────────────────────────────────
* checking for file ‘/tmp/RtmpKXD6jd/remotes142571188a0/fhernanb-stests-a01b9bf/DESCRIPTION’ ... OK
* preparing ‘stests’:
* checking DESCRIPTION meta-information ... OK
* checking for LF line-endings in source and make files and shell scripts
* checking for empty or unneeded directories
Omitted ‘LazyData’ from DESCRIPTION
* building ‘stests_0.1.0.tar.gz’



Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



**Comparación de Varianzas de Distancias de Vuelo entre Aeropuertos JFK y LGA**


**PROBLEMA**

Con el objetivo de mejorar la eficiencia en la asignación de aeronaves y recursos, el equipo de logística de una aerolínea desea comparar la variabilidad en las distancias de vuelo de los aeropuertos JFK (John F. Kennedy) y LGA (LaGuardia).

Según informes previos, se sospecha que la variabilidad de la distancia de los vuelos de cada aeropuerto es similar. Por lo tanto, se desea realizar una prueba de hipótesis para determinar si se puede suponer que la varianza de las distancias de los vuelos de JFK es igual a la de LGA, con un nivel de significancia del 5%.

Se asume que ambas poblaciones siguen una distribución normal, y se toma JFK como
𝑚
 y LGA como
𝑛
.


Sabiendo que:


 $H_0: \frac{\sigma_m^2}{\sigma_n^2} = 1$


 $H_1: \frac{\sigma_m^2}{\sigma_n^2} \neq  1$



**Tipo de prueba:** Dos colas de distribución F.



Para validar dicha afirmación, se procede a realizar la prueba en R:

Para investigar esta diferencia, se recopilan dos muestras aleatorias de 20 vuelos con reemplazo desde una base de datos, seleccionando la variable distance de la base de datos flights del paquete nycflights13, que representa la distancia recorrida por cada vuelo en millas.





In [ ]:
# 1. Fijar semilla para reproducibilidad
set.seed(1000)

# 2. Filtrar vuelos según el origen JFK y LGA
JFK.flights <- subset(flights, origin == "JFK")   # Todos los vuelos desde JFK
LGA.flights <- subset(flights, origin == "LGA")   # Todos los vuelos desde LGA

# 3. Tomar una muestra aleatoria de 20 vuelos de cada uno con reemplazo
sample.JFK <- JFK.flights[sample(1:nrow(JFK.flights), 20, replace = T), ]
sample.LGA <- LGA.flights[sample(1:nrow(LGA.flights), 20, replace = T), ]

#4. Calcular el resultado
resultado.varianzas <- stests::var.test(sample.JFK$distance, sample.LGA$distance, alternative = "two.sided", conf.level= 0.95)

# 5. Se realiza la prueba de hipótesis
cat("===== Prueba de hipótesis del COCIENTE DE VARIANZAS de los vuelos =====\n")
print(resultado.varianzas)

===== Prueba de hipótesis del COCIENTE DE VARIANZAS de los vuelos =====

	F test to compare two variances

data:  sample.JFK$distance  and  sample.LGA$distance
F = 2.461, num df = 19, denom df = 19, p-value = 0.05663
alternative hypothesis: true ratio of variances is not equal to 1
95 percent confidence interval:
 0.9740873 6.2175547
sample estimates:
ratio of variances 
          2.460984 



**ANÁLISIS DEL COCIENTE DE VARIANZAS**

**Resumen de la prueba**

Valor observado: $F_{\text{obs}} = 2.461$

Grados de libertad: JFK = 19, LGA = 19

Estadístico de prueba: $\frac{S_m^2}{S_n^2} = 2.460984 $

Región de aceptación: $]0.9740873,\ 6.2175547[$

Región de rechazo: $]0,\ 0.9740873[ \cup ]6.2175547,\ \infty[$

Nivel de significancia: $\alpha = 0.05$

Nivel de confianza: $0.95$

Valor P de la prueba: $0.05663$



 **Conclusión**

Dado que el valor 1 (hipótesis nula) se encuentra dentro del intervalo de aceptación y el valor-p = 0.05663 es mayor que
𝛼
=
0.05, no se rechaza la hipótesis nula.

Por lo tanto, no se encontró evidencia estadísticamente significativa para afirmar que las varianzas son diferentes. Se puede asumir igualdad de varianzas entre las distancias de vuelo de los aeropuertos JFK y LGA, respaldando así la suposición inicial del equipo logístico.

# **Prueba de hipotesis para 2 poblaciones utilizando T-Student**

**Problema:**

Con el fin de optimizar la asignación de aeronaves según la duración promedio de los vuelos, el equipo de planificación de rutas de una aerolínea desea investigar si existe una diferencia significativa en la distancia promedio de los vuelos que parten del aeropuerto JFK (John F. Kennedy) respecto a los del LGA (LaGuardia), ambos en la ciudad de Nueva York.

Aunque se sabe que los destinos de ambos aeropuertos pueden variar, se sospecha que las distancias promedio son similares, lo cual permitiría asignar los mismos tipos de aeronaves a ambos aeropuertos. Para confirmar esta suposición, se decide realizar una prueba de hipótesis con un nivel de significancia del 5%.

**Sabiendo que:**

* μ₁ = media poblacional de las distancias desde JFK

* μ₂ = media poblacional de las distancias desde LGA

* H₀: μ₁ = μ₂ (las distancias promedio desde JFK y LGA son iguales).

* H₁: μ₁ ≠ μ₂ (las distancias promedio desde JFK y LGA son diferentes).
* Las dos muestras son independientes.

* Las poblaciones tienen distribución normal.

* Según el analisis de conciente de varianzas realizado previamente, se puede asumir que las varianzas son iguales (p > 0.05).

In [ ]:
# 1. Prueba t para diferencia de medias, asumiendo varianzas iguales
resultado.medias <- t.test(sample.JFK$distance, sample.LGA$distance,
                           var.equal = TRUE, alternative = "two.sided", conf.level = 0.95)

# 2. Mostrar resultado
cat("===== Prueba T para Diferencia de Medias entre JFK y LGA =====\n")
print(resultado.medias)

===== Prueba T para Diferencia de Medias entre JFK y LGA =====

	Two Sample t-test

data:  sample.JFK$distance and sample.LGA$distance
t = 0.061187, df = 38, p-value = 0.9515
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -364.1678  386.8678
sample estimates:
mean of x mean of y 
   921.85    910.50 



### **Resumen de prueba:**
* Valor observado	$t_{\text{obs}}$ = 0.0612
* Grados de libertad =	38
* Estadístico de prueba	x̄ = 921.85 (JFK)- 910.50 (LGA) = 11.35
* Región de aceptación = ]-364.1678, 386.8678[
* Región de rechazo	t = ]−∞, -364.1678[ ∪ ]386.8678, ∞[
* Significancia estadística	α =  0.05
* Nivel de confianza = 0.95
* Valor P de la prueba = 0.9515

**Conclusión:**

Dado que el valor p = 0.9515 es mucho mayor que el nivel de significancia α = 0.05, no existe evidencia estadística suficiente para rechazar la hipótesis nula H₀. Es decir, no se puede afirmar que haya una diferencia significativa entre la distancia promedio de los vuelos que parten desde los aeropuertos JFK y LGA.

**UNA VARIANZA**



**PROBLEMA:**

Una compañía desea conocer la variabilidad de las distancias de vuelo desde el aeropuerto EWR, ya que esto influye en la logística de operaciones y asignación de recursos. Según sus estimaciones internas, se afirma que la varianza poblacional es menor que 553 200 a un nivel de significancia de 5%, suponiendo que la población tiene una distribución normal. Se desea hacer una prueba de hipótesis para verificar lo dicho por la compañía.




Sabiendo que:


 $H_0: {\sigma^2} = 553 200$ ($\geq$)


 $H_1: \sigma^2 < 553 200$



**Tipo de prueba:** Prueba $X^2$ de cola inferior.



Para validar dicha afirmación, se procede a realizar la prueba en R:

Para validar esta afirmación, se toma una muestra aleatoria de 30 vuelos desde la base de datos flights del paquete nycflights13. Se analiza la variable distance, que representa la distancia recorrida en millas por cada vuelo.

In [ ]:
# 1. Fijar semilla para reproducibilidad
set.seed(1000)

# 2. Filtrar vuelos según el origen EWR
EWR.flights <- subset(flights, origin == "EWR")   # Todos los vuelos desde EWR

# 3. Tomar una muestra aleatoria de 30 con reemplazo
sample.EWR <- EWR.flights[sample(1:nrow(EWR.flights), 30, replace = T), ]

resultado.varianza <-stests::var.test(sample.EWR$distance, null.value = 553200, alternative = "less", conf.level = 0.95)

#Utilización de la función varTest para calculo de la prueba de hipótesis
cat("===== Prueba de hipótesis para la Varianza =====\n")
print(resultado.varianza)

===== Prueba de hipótesis para la Varianza =====

	X-squared test for variance

data:  sample.EWR$distance
X-squared = 18.241, df = 29, p-value = 0.06079
alternative hypothesis: true variance is less than 553200
95 percent confidence interval:
 237119.8      Inf
sample estimates:
variance of x 
       347969 



**ANÁLISIS DEL INTERVALO DE CONFIANZA DE LA VARIANZA**

**Resumen de la prueba**

Valor observado: $X^2_{\text{obs}} = 18.241$

Grados de libertad:   $29$

Estadístico de prueba: $S^2$ = 347969

Región de aceptación: $]237119.8 ,\ \infty[$

Región de rechazo: $] 0, 237119.8 [$

Nivel de significancia: $\alpha = 0.05$

Nivel de confianza: $0.95$

Valor P de la prueba: $ 0.06079$.



**Conclusion:**

Dado que el valor p = 0.06079 es mayor que el nivel de significancia 𝛼 = 0.05, y que el valor 553 200 se encuentra dentro del intervalo de aceptación
[237 119.8, ∞[, no se rechaza la hipótesis nula  $H_0$.
Por lo tanto, no existe evidencia estadística suficiente para concluir que la varianza poblacional de las distancias de vuelo desde el aeropuerto EWR sea menor que 553 200. En consecuencia, los datos no respaldan la afirmación realizada por la compañía.

**UNA PROPORCION**

**Proporción de Vuelos de la Aerolínea United Airlines (UA)**

**PROBLEMA**

El equipo de logística de la aerolínea desea conocer si más del 15% de los vuelos registrados en el aeropuerto JFK corresponden a la aerolínea United Airlines (código "UA") para justificar nuevas alianzas estratégicas.

Se desea realizar una prueba de hipótesis para verificar si la proporción de vuelos de la aerolínea UA es mayor al 15% en JFK. Para ello, se toma una muestra aleatoria de 300 vuelos desde JFK utilizando la base de datos flights del paquete nycflights13.

Se realizará una prueba de hipótesis para una proporción con un nivel de significancia del 5%, se asume normalidad en la distribucion de la poblacion.

**Planteamiento de la hipótesis**

Sea
𝑝
 la proporción de vuelos de la aerolínea UA desde el aeropuerto JFK.

Hipótesis nula:
$H_0:$
𝑝
=
0.15

Hipótesis alternativa:
$H_1:$
𝑝 > 0.15

**Tipo de prueba:** una cola hacia la derecha.



In [ ]:
# 1. Cargar librerías necesarias
library(nycflights13)

# 2. Fijar semilla para reproducibilidad
set.seed(2025)

# 3. Filtrar vuelos desde JFK y eliminar NA en 'carrier'
JFK.flights <- subset(flights, origin == "JFK" & !is.na(carrier))

# 4. Tomar una muestra aleatoria de 300 vuelos (con reemplazo)
muestra.JFK <- JFK.flights[sample(1:nrow(JFK.flights), 300, replace = TRUE), ]

# 5. Contar la cantidad de vuelos de UA en la muestra
num.UA <- sum(muestra.JFK$carrier == "UA")

# 6. Realizar prueba de proporción (sin corrección de continuidad)
resultado.proporcion <- prop.test(num.UA, 300, p = 0.15, alternative = "greater", correct = FALSE)

# 7. Mostrar resultados
cat("===== Prueba de hipótesis para la PROPORCIÓN de vuelos UA desde JFK =====\n")
print(resultado.proporcion)

===== Prueba de hipótesis para la PROPORCIÓN de vuelos UA desde JFK =====

	1-sample proportions test without continuity correction

data:  num.UA out of 300, null probability 0.15
X-squared = 20.497, df = 1, p-value = 1
alternative hypothesis: true p is greater than 0.15
95 percent confidence interval:
 0.03841471 1.00000000
sample estimates:
         p 
0.05666667 



**Análisis de la Prueba de Hipótesis para la Proporción de Vuelos de UA desde JFK**

Valor observado:
$Z_{\text{obs}}$
=
$\sqrt{20.497} \approx 4.527$


Grados de libertad: NA

Estadístico de prueba:
$\hat{p}$
=
0.0567


Región de aceptación: [0.03841471, 1.00000000]

Región de rechazo: [0, 0.03841471]

Nivel de significancia:
𝛼
=
0.05


Nivel de confianza: 0.95

Valor P de la prueba: 1

**Conclusión**

Se realizó una prueba de hipótesis para determinar si la proporción de vuelos operados por la aerolínea United Airlines (UA) desde el aeropuerto JFK en 2013 era mayor al 15%, basándose en una muestra aleatoria de 300 vuelos.

El resultado del test muestra un valor p igual a 1, lo cual es extremadamente alto y sugiere que no hay ninguna evidencia estadística para rechazar la hipótesis nula. Además, el intervalo de confianza del 95% obtenido va de 3.84% a 100%, conteniendo el valor de 15%. Esto muestra que no podemos afirmar que la proporción real supere el 15%.

Por lo tanto, no se rechaza la hipótesis nula. Los datos recolectados no respaldan la idea de que UA opere más del 15% de los vuelos desde JFK.


**Diferencia de Proporciones: Vuelos Puntuales UA vs B6**

**PROBLEMA**

La puntualidad es un aspecto fundamental para los pasajeros al momento de elegir una aerolínea. Un estudio previo consultó a usuarios de vuelos en Nueva York sobre cuál aerolínea consideraban más puntual entre United Airlines (UA) y JetBlue Airways (B6), y la mayoría afirmó que B6 era más puntual que UA.

Con base en esa percepción, un grupo de analistas decide evaluar si realmente existe una diferencia significativa en la puntualidad entre ambas aerolíneas. Para ello, se toma una muestra aleatoria de 100 vuelos de cada aerolínea con reemplazo a partir de los datos del año 2013. Se define como vuelo puntual aquel cuya llegada tuvo un retraso menor o igual a 15 minutos.

Luego de identificar la cantidad de vuelos puntuales en cada grupo, se realiza una prueba de hipótesis para diferencia de proporciones, con un intervalo de confianza del 95%, para determinar si la diferencia observada es estadísticamente significativa, la distribucion se asume normal para ambas poblaciones.

**Planteamiento de la hipótesis**

Sea
$p_1$ la proporción de vuelos puntuales de UA, y $p_2$ la de B6.

Hipótesis nula:
$H_0:$
$p_1$
=
$p_2$

​
 Hipótesis alternativa:
$H_1:$
$p_1$
≠
$p_2$

**Tipo de prueba:** Dos colas (no se asume cuál es mayor).

In [ ]:
library(nycflights13)
library(dplyr)

# Fijar semilla para reproducibilidad
set.seed(1000)

# Filtrar vuelos válidos de UA y B6 (eliminando NA en arr_delay)
ua <- flights %>% filter(carrier == "UA" & !is.na(arr_delay))
b6 <- flights %>% filter(carrier == "B6" & !is.na(arr_delay))

# Tomar muestras aleatorias de 100 vuelos con reemplazo
muestra_ua <- ua %>% sample_n(100, replace = TRUE)
muestra_b6 <- b6 %>% sample_n(100, replace = TRUE)

# Convertir condición de puntualidad a variable binaria: 1 si puntual, 0 si no
muestra_ua$puntual <- ifelse(muestra_ua$arr_delay <= 15, 1, 0)
muestra_b6$puntual <- ifelse(muestra_b6$arr_delay <= 15, 1, 0)

# Contar vuelos puntuales con table()
table_ua <- table(muestra_ua$puntual)
table_b6 <- table(muestra_b6$puntual)

# Éxitos (puntuales = 1)
x_ua <- table_ua["1"]
x_b6 <- table_b6["1"]

# Prueba de proporciones
prop.test(c(x_ua, x_b6), c(100, 100))



	2-sample test for equality of proportions with continuity correction

data:  c(x_ua, x_b6) out of c(100, 100)
X-squared = 0.030139, df = 1, p-value = 0.8622
alternative hypothesis: two.sided
95 percent confidence interval:
 -0.102864  0.142864
sample estimates:
prop 1 prop 2 
  0.80   0.78 


**Análisis de la prueba de diferencia de proporciones**

Se realizó una prueba de hipótesis para comparar la proporción de vuelos puntuales (definidos como aquellos con un retraso menor o igual a 15 minutos) entre United Airlines (UA) y JetBlue Airways (B6).

**Los resultados de la prueba fueron los siguientes:**

**Resumen de la prueba**


Valor observado:
$Z_{\text{obs}}$
=
$\sqrt{0.0301} \approx 0.1734$


Grados de libertad: NA

Estadístico de prueba:
$\hat{p}_1$
=
0.80

$\hat{p}_2$
=
0.78


Región de aceptación: [ -0.1029 , 0.1429 ]

Región de rechazo:
]
−
1
,
−
0.1029
[
∪
]
0.1429
,
1
[


Nivel de significancia:
𝛼=
0.05

Nivel de confianza: 0.95

Valor P de la prueba: 0.8622



**Conclusión**

El valor p obtenido (0.8622) es mucho mayor que el nivel de significancia habitual de 0.05, lo que significa que no hay evidencia suficiente para rechazar la hipótesis nula. Además, el intervalo de confianza del 95% para la diferencia de proporciones incluye el valor 0, no se puede concluir que exista una diferencia estadísticamente significativa entre ambas aerolíneas..

A partir de la muestra analizada, no se encontró evidencia estadística significativa para afirmar que existe una diferencia real en la puntualidad entre United Airlines (UA) y JetBlue Airways (B6). Por tanto, los datos no respaldan la percepción general de que B6 es significativamente más puntual que UA.

## **Promedio de Retraso de Salida en el Aeropuerto LGA Usando Distribución Z**
### **Problema**

El equipo de operaciones del aeropuerto LaGuardia (LGA) quiere evaluar si el promedio de retraso en salidas de sus vuelos supera los 10 minutos, lo cual es el estándar considerado aceptable por la gerencia para una operación eficiente.

Se supone que el retraso en salida tiene una distribución normal y se conoce la desviación estándar muestral. Para verificar, se toma una muestra aleatoria de vuelos y se aplica una prueba Z para el promedio, con un nivel de confianza del 95%. Se desea hacer una prueba de hipótesis para verificar lo dicho por la compañía.

Sabiendo que:

H₀: μ = 10 minutos (≤)

H₁: μ > 10 minutos

**Tipo de prueba:** Z de cola derecha (unilateral superior)

Para validar dicha afirmación, se procede a realizar la prueba en R:

Para investigar el retraso promedio de las salidas de vuelos desde el aeropuerto LGA, se recopila una muestra aleatoria de 40 vuelos con reemplazo desde una base de datos, seleccionando la variable dep_delay, que representa el retraso de salida en minutos. Bajo el supuesto de normalidad en la distribución de los datos y con una desviación estándar muestral conocida, se aplica una prueba Z para una media con un nivel de confianza del 95%.






In [ ]:
#Instalación del paquete stests
if (!require('devtools')) install.packages('devtools')
devtools::install_github('fhernanb/stests', force=TRUE)


── R CMD build ─────────────────────────────────────────────────────────────────
* checking for file ‘/tmp/Rtmpaa1njG/remotes1198d3146e/fhernanb-stests-a01b9bf/DESCRIPTION’ ... OK
* preparing ‘stests’:
* checking DESCRIPTION meta-information ... OK
* checking for LF line-endings in source and make files and shell scripts
* checking for empty or unneeded directories
Omitted ‘LazyData’ from DESCRIPTION
* building ‘stests_0.1.0.tar.gz’



Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



In [ ]:
set.seed(1000)

# Filtrar vuelos del aeropuerto LGA y eliminar NA en dep_delay
LGA.flights <- subset(flights, origin == "LGA" & !is.na(dep_delay))

# Tomar una muestra aleatoria de 40 vuelos con reemplazo
sample.LGA <- LGA.flights[sample(1:nrow(LGA.flights), 40, replace = TRUE), ]

# Calcular la desviación estándar muestral
std_dev_dep_delay <- sd(sample.LGA$dep_delay)
# Aplicar la prueba Z
resultado <- stests::z.test(
  x = sample.LGA$dep_delay,
  sigma2 = std_dev_dep_delay,
  mu = 10,
  conf.level = 0.95,
  alternative = "greater"
)

cat("===== Prueba de hipótesis de 95% para el Promedio de Retraso =====\n")
print(resultado)

===== Prueba de hipótesis de 95% para el Promedio de Retraso =====

	One Sample z-test

data:  sample.LGA$dep_delay
Z = -5.3143, p-value = 1
alternative hypothesis: true mean is greater than 10
95 percent confidence interval:
 4.532777      Inf
sample estimates:
mean of sample.LGA$dep_delay 
                       5.825 



## **Analisis del promedio usando distribución Z**
**Resumen de la prueba**

Valor observado:  $Z_{\text{obs}}$ = -5.3143

Grados de libertad:  NA (no aplica para Z)

Estadístico de prueba:  x̄ = 5.825

Región de aceptación:  ]4.532777, ∞[

Región de rechazo: ]-∞, 4.532777]

Nivel de significancia:  α=0.05

Nivel de confianza:  0.95

Valor P de la prueba:  1

**Conclusión:**

Dado que el valor-p = 1 es mayor que α = 0.05, no se rechaza la hipótesis nula. Además, el valor 10 cae dentro de la región de aceptación. Por lo tanto, no hay evidencia estadística suficiente para afirmar que el retraso promedio en las salidas de vuelos desde el aeropuerto LaGuardia (LGA) sea mayor a 10 minutos.




# **Prueba de hipótesis para una población con la distribución T Student**

### **Problema:**
Una aerolínea afirma que el tiempo promedio de vuelo de todos sus vuelos que parten desde el aeropuerto John F. Kennedy (JFK) es de 150 minutos. Con el fin de evaluar la veracidad de esta afirmación, un equipo de analistas decide realizar un estudio estadístico. Para ello, se selecciona una muestra aleatoria representativa de vuelos que han salido recientemente desde el aeropuerto JFK y se registra la duración de cada uno de ellos. Posteriormente, se lleva a cabo una prueba de hipótesis sobre la media poblacional para determinar si existe evidencia suficiente para rechazar la afirmación de la aerolínea. El análisis se realiza utilizando un nivel de significancia del 5% .

### **Hipotesis:**
**Sabiendo que:**

* H₀: μ = 150

* H₁: μ ≠ 150

* Tipo de prueba: dos colas

* Distribución: T de Student

* Condiciones asumidas:

  * Muestra aleatoria.

  * La variable air_time sigue aproximadamente una distribución normal.

  * La desviación estándar poblacional es desconocida.

In [ ]:
# Filtramos datos para JFK y eliminamos NA en air_time
jfk_airtimes <- flights %>%
  filter(origin == "JFK" & !is.na(air_time)) %>%
  select(air_time)

# Seleccionamos una muestra aleatoria de 20 vuelos
set.seed(123)
muestra <- sample(jfk_airtimes$air_time, size = 20)

t.test(muestra, mu = 150, alternative = "two.sided", conf.level = 0.95)


	One Sample t-test

data:  muestra
t = 1.1967, df = 19, p-value = 0.2462
alternative hypothesis: true mean is not equal to 150
95 percent confidence interval:
 128.39 229.31
sample estimates:
mean of x 
   178.85 


### **Resumen de prueba:**
* Valor observado	$t_{\text{obs}}$ = 1.1967
* Grados de libertad =	19
* Estadístico de prueba	x̄ = 178.85
* Región de aceptación = ] 128.39, 229.31[
* Región de rechazo	t = ]−∞, 128.39[ ∪ ]229.31, ∞[
* Significancia estadística	α =  0.05
* Nivel de confianza = 0.95
* Valor P de la prueba = 0.2462

### **Conclusión:**
Debido a que el valor propuesto de que el timepo promedio de vuelos desde (JFK) es de 150 minutos y se encuentra dentro del intervalo de aceptación ] 128.39, 229.31[ y que el valor P = 0.2462 es mayor que α = 0.05, no se cuenta con evidencia estadísticamente significativa para rechazar la hipótesis nula H₀. Por lo tanto, no se puede concluir que el tiempo promedio de vuelo desde el aeropuerto JFK sea diferente de 150 minutos, y se mantiene la afirmación de la aerolínea con el nivel de confianza del 95%.

# **Prueba de Hipótesis para dos poblaciones con la distribución z**

**Problema:**

Los meteorólogos quieren saber si la temperatura promedio en julio supera significativamente a la de enero en Nueva York. Se asume que las temperaturas de ambos meses siguen una distribución normal. Para ello, se toma una muestra aleatoria de datos de temperatura diaria de cada mes y, con un nivel de significancia del 5 %, se realiza una prueba de hipótesis de dos muestras independientes.

**Hipótesis:**

Sabiendo que:

H₀: μ_Julio – μ_Enero = 0

H₁: μ_Julio – μ_Enero > 0

**Tipo de prueba:**

Cola derecha(temperatura de Julio mayor que la de Enero)
Nivel de significancia: α = 0.05

**Distribución:**

Z (normal estándar)

**Condiciones asumidas:**

Muestras aleatorias e independientes.

La variable “temperatura” sigue aproximadamente una distribución normal en cada mes.

Desviaciones estándar muestrales conocidas.


In [ ]:
install.packages("PASWR2")
library(PASWR2)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)

also installing the dependencies ‘proxy’, ‘e1071’


Loading required package: lattice

Loading required package: ggplot2



In [ ]:
# 1. Fijar semilla para reproducibilidad
set.seed(2025)

# 2. Filtrar datos de temperaturas para Julio (mes 7) y Enero (mes 1)
data_jul <- subset(weather, month == 7 & !is.na(temp))$temp
data_jan <- subset(weather, month == 1 & !is.na(temp))$temp

# 3. Tomar muestra aleatoria de 30 observaciones con reemplazo
sample_jul <- sample(data_jul, 30, replace = TRUE)
sample_jan <- sample(data_jan, 30, replace = TRUE)

# 4. Estimamos las desviaciones
sigma_jul <- sd(sample_jul)
sigma_jan <- sd(sample_jan)

# 5. Prueba Z (greater)
z_result <- PASWR2::z.test(x = sample_jul,
                   y = sample_jan,
                   sigma.x = sigma_jul,
                   sigma.y = sigma_jan,
                   alternative = "greater",
                   mu = 0,
                   conf.level = 0.95)

# 6. Mostrar resumen
print("===== Resultados de la prueba Z (Julio > Enero) =====")
print(z_result)


[1] "===== Resultados de la prueba Z (Julio > Enero) ====="

	Two Sample z-test

data:  sample_jul and sample_jan
z = 19.462, p-value < 2.2e-16
alternative hypothesis: true difference in means is greater than 0
95 percent confidence interval:
 42.94911      Inf
sample estimates:
mean of x mean of y 
   80.288    33.374 



**Resumen de la prueba:**

Valor observado: $Z_{\text{obs}}$ = 19.462

Grados de libertad: NA

Estadístico de prueba: $x̂_{\text{julio}}$ = 80.288 - $x̂_{\text{enero}}$ = 33.374 = 46.914


Región de aceptación: $] 42.94911 ,\ \infty[$


Región de rechazo: $] \ -\infty, 42.94911 [$

Significancia estadística : 0.05

Nivel de confianza: 0.95

p-valor = 2.2 × 10⁻¹⁶ (aprox.)



**Conclusión:**

Dado que el valor 0 no se encuentra en el intervalo de aceptación y el p-valor es prácticamente cero, se rechaza la hipótesis nula H₀. Hay evidencia estadística fuerte de que la temperatura promedio en julio es significativamente mayor que en enero en Nueva York.

**OPCIONES INVESTIGADAS PARA LAS DISTRIBUCIONES Z**





**Paquete: BSDA**


z.test(x, y = NULL, alternative = "two.sided", mu = 0, sigma.x = NULL, sigma.y = NULL, conf.level = 0.95)

Parámetros:
x: vector numérico.
y: vector numérico opcional.
alternative: define la hipótesis alternativa ("two.sided", "less", "greater").
mu: valor bajo la hipótesis nula.
sigma.x: desviación estándar poblacional para x.
sigma.y: desviación estándar poblacional para y.
conf.level: nivel de confianza del intervalo.
Aplica para:
Una población
Dos poblaciones

**Paquete: stests**


z.test(x, sigma2, alternative = "two.sided", mu = 0, conf.level = 0.95)

Parámetros:
x: vector numérico.
sigma2: varianza poblacional conocida.
alternative: define la hipótesis alternativa ("two.sided", "less", "greater").
mu: valor bajo la hipótesis nula.
conf.level: nivel de confianza del intervalo.
Aplica para:
Una población


**Paquete: PASWR2**


z.test(x, sigma.x = NULL, y = NULL, sigma.y = NULL, sigma.d = NULL, alternative = c("two.sided", "less", "greater"), mu = 0, paired = FALSE, conf.level = 0.95, ...)

Parámetros:
x: vector numérico.
sigma.x: desviación estándar poblacional de x.
y: vector numérico opcional.
sigma.y: desviación estándar poblacional de y.
sigma.d: desviación estándar poblacional de las diferencias (pareadas).
alternative: define la hipótesis alternativa ("two.sided", "less", "greater").
mu: valor bajo la hipótesis nula.
paired: indica si se trata de una prueba pareada.
conf.level: nivel de confianza del intervalo.
Aplica para:
Una población
Dos poblaciones


In [ ]:
mi_dataframe <- data.frame(x = 1:100, y = paste0("valor_", 1:100))

**ANÁLISIS DE LA PRUEBA DE DIFERENCIA DE MEDIAS (Z, cola derecha)**

**Resumen de la prueba**

* **Valor observado**:

  $$
    Z_{\rm obs} = 19.462
  $$
* **Grados de libertad**: NA
* **Estadístico de prueba**:

  $$
    \bar x_{Julio} - \bar x_{Enero} = 80.288 - 33.374 = 46.914
  $$
* **Región de aceptación** (en la escala de la diferencia de medias):

  $$
    [\,-\infty,\;42.94911\,]
  $$
* **Región de rechazo** (cola derecha):

  $$
    ]\,42.94911,\;\infty[
  $$
* **Nivel de significancia**:
  $\alpha = 0.05$
* **Nivel de confianza**:
  $1-\alpha = 0.95$
* **p‑valor**:

  $$
    P(Z>19.462)\approx 2.2\times10^{-16}
  $$

---

**Análisis:**
Para esta prueba se comparó la temperatura diaria promedio de julio frente a la de enero en Nueva York, usando muestras independientes de $n=30$ observaciones (con reemplazo) en cada mes. Se generaron así dos vectores de datos (`sample_jul` y `sample_jan`) tras fijar semilla para reproducibilidad.

La media de julio resultó $\bar x_{Julio}=80.288$ y la de enero $\bar x_{Enero}=33.374$, con una diferencia muestral de $46.914$. Las desviaciones estándar muestrales se estimaron como $\sigma_{Julio}=sd(sample\_jul)$ y $\sigma_{Enero}=sd(sample\_jan)$. A partir de ellas se calculó el error estándar

$$
  SE = \sqrt{\frac{\sigma_{Julio}^2}{30} + \frac{\sigma_{Enero}^2}{30}}
$$

y el estadístico

$$
  Z_{\rm obs}
  = \frac{(\bar x_{Julio}-\bar x_{Enero}) - 0}{SE}
  = 19.462.
$$

Con $\alpha=0.05$ y prueba de cola derecha, el valor crítico en Z es $Z_{0.95}=1.645$, lo que equivale en la escala de diferencias a $42.94911$. Dado que la diferencia observada $46.914$ cae en la región de rechazo $\bigl]42.94911,\infty\bigr[$ y el p‑valor (\~$2.2\times10^{-16}$) es muchísimo menor que 0.05, **rechazamos la hipótesis nula**.

En conclusión, existe evidencia estadística muy fuerte de que la temperatura promedio en julio es significativamente mayor que en enero en Nueva York (nivel de confianza 95 %).
